File: src/main/java/uk/gov/justice/digital/hmpps/oauth2server/nomis/repository/StaffUserAccountRepository.java
package uk.gov.justice.digital.hmpps.oauth2server.nomis.repository;

import org.springframework.data.repository.CrudRepository;
import uk.gov.justice.digital.hmpps.oauth2server.nomis.model.NomisUserPersonDetails;
import uk.gov.justice.digital.hmpps.oauth2server.security.UserPersonDetails;

import java.util.List;

public interface StaffUserAccountRepository extends CrudRepository<NomisUserPersonDetails, String> {
    List<NomisUserPersonDetails> findByStaffFirstNameIgnoreCaseAndStaffLastNameIgnoreCase(String firstName, String lastName);
}

File: src/main/java/uk/gov/justice/digital/hmpps/oauth2server/nomis/repository/StaffUserAccountRepository.kt
package uk.gov.justice.digital.hmpps.oauth2server.nomis.repository

import org.springframework.data.jpa.repository.Query
import org.springframework.data.repository.CrudRepository
import uk.gov.justice.digital.hmpps.oauth2server.nomis.model.NomisUserPersonDetails

@Suppress("SqlResolve")
interface StaffUserAccountRepository : CrudRepository<NomisUserPersonDetails, String> {
  @Query(value = """select distinct s.*, '' as SPARE4 from STAFF_USER_ACCOUNTS s
    inner join INTERNET_ADDRESSES i on i.owner_id = s.staff_id and owner_class = 'STF'
    where internet_address_class = 'EMAIL' and i.internet_address = ?""", nativeQuery = true)
  fun findAllNomisUsersByEmailAddress(email: String): List<NomisUserPersonDetails>

  fun findByStaffFirstNameIgnoreCaseAndStaffLastNameIgnoreCase(firstName: String, lastName: String): List<NomisUserPersonDetails>
}
